main.go

```diff
@@ -214,33 +210,45 @@ func setupControllers(ctx context.Context, mgr ctrl.Manager, cCache *cache.Cache
 		os.Exit(1)
 	}
 
-	crds := findCustomResources(ctx, mgr)
-
 	opts := []jobframework.Option{
 		jobframework.WithManageJobsWithoutQueueName(manageJobsWithoutQueueName),
 		jobframework.WithWaitForPodsReady(waitForPodsReady(cfg)),
 		jobframework.WithKubeServerVersion(serverVersionFetcher),
 	}
 	err := jobframework.ForEachIntegration(func(name string, cb jobframework.IntegrationCallbacks) error {
 		log := setupLog.WithValues("jobFrameworkName", name)
-		if isFrameworkEnabled(cfg, name) && crds.Has(name) {
-			if err := cb.NewReconciler(
+
+		_, err := mgr.GetRESTMapper().RESTMapping(cb.GVK.GroupKind(), cb.GVK.Version)
+		if err != nil {
+			// TODO: If the below PR is released, we need to change a way to check if the GVK is registered.
+			// REF: https://github.com/kubernetes-sigs/controller-runtime/pull/2425
+			// if !meta.IsNoMatchError(err) {
+			//   return err
+			// }
+			// log.Info("No matching API server for job framework, skip to create controller and webhook")
```

Comment:
Since the controller-runtime v0.15, the returned error seems to be changed with discovery.ErrGroupDiscoveryFailed .

And the changing seems the unintended breaking changes:
https://github.com/kubernetes-sigs/controller-runtime/pull/2425